teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,369 @@
+tracked artifact, this process aims to enhance communication and
+discoverability.
+
+A TEP is broken into sections which can be merged into source control
+incrementally in order to support an iterative development process. An
+important goal of the TEP process is ensuring that the process for
+submitting the content contained in [design proposals][] is both clear
+and efficient. The TEP process is intended to create high quality
+uniform design and implementation documents for WGs to deliberate.
+
+[road to Go 2]: https://blog.golang.org/toward-go2) 
+[design proposals]: https://github.com/kubernetes/community/tree/master/contributors/design-proposals
+
+## Stewardship
+The following DACI model indentifies the responsible parties for TEPs:
+
+**Workstream** | **Driver** | **Approver** | **Contributor** |
+**Informed** --- | --- | --- | --- | ---
+| TEP Process Stewardship | TG members | TG members | TG members | Community |
+| Enhancement delivery | Enhancement Owner | SIG leadership (SIG Chairs + TLs) | Enhancement Implementer(s) (may overlap with Driver) | Community |
+
+*TG members: Tekton Governing board members*
+
+## Reference-level explanation
+
+### What type of work should be tracked by a TEP
+
+The definition of what constitutes an "enhancement" is a foundational
+concern for the Tekton project. Roughly any Tekton user or
+operator facing enhancement should follow the TEP process. If an
+enhancement would be described in either written or verbal
+communication to anyone besides the TEP author or developer, then
+consider creating a TEP.
+
+Similarly, any technical effort (refactoring, major architectural
+change) that will impact a large section of the development community
+should also be communicated widely. The TEP process is suited for this
+even if it will have zero impact on the typical user or operator.
+
+As the local bodies of governance, WGs should have broad latitude in
+describing what constitutes an enhancement which should be tracked
+through the TEP process. WGs may find it helpful to enumerate what
+_does not_ require a TEP rather than what does. WGs also have the
+freedom to customize the TEP template according to their WG specific
+concerns. For example, the TEP template used to track API changes will
+likely have different subsections than the template for proposing
+governance changes. However, as changes start impacting other WGs or
+the larger developer communities outside of a WG, the TEP process
+should be used to coordinate and communicate.
+
+Enhancements that have major impacts on multiple WGs should use the
+TEP process. A single WG will own the TEP but it is expected that
+the set of approvers will span the impacted WGs. The SEP process is
+the way that WGs can negotiate and communicate changes that cross
+boundaries.
+
+TEPs will also be used to drive large changes that will cut across all
+parts of the project. These TEPs will be owned by SIG-architecture
```

Comment:
? 
We don't have a cross-project Tekton technical oversight body right? Do we need one?
Thinking about all cross-project type of activities:
- Infra like release, CI and CD can be supported by the Productivity WG. 
- Documentation is supported by the Docs WG. 
- API strategies and stability we have the Beta WG, which partly covers for that, and might evolve into an API WG once we reach beta. Or perhaps it will become a GA WG :)
- From the operational POV i.e. consistent install/upgrade, logging, monitoring across projects - which we don't have ATM, but that might also be covered by a dedicated WG if we feel the need for it.

Reply:
Indeed we don't. So far, if it's cross-project, my feeling was to use the *main* WG for this. But yeah, we might need to think more about that.